message_definitions/v1.0/ras_a.xml

```diff
@@ -0,0 +1,124 @@
+      <entry value="4" name="POI_CLASSIFICATION_BUILDING">
+        <description>Building.</description>
+      </entry>
+      <entry value="5" name="POI_CLASSIFICATION_PERSON">
+        <description>Person.</description>
+      </entry>
+      <entry value="6" name="POI_CLASSIFICATION_VEHICLE">
+        <description>Vehicle.</description>
+      </entry>
+      <entry value="7" name="POI_CLASSIFICATION_INGRESS_PORTAL">
+        <description>Ingress portal.</description>
+      </entry>
+    </enum>
+  </enums>
+  <messages>
+    <message id="238" name="POI_REPORT">
+      <wip/>
+      <!-- This message is work-in-progress and it can therefore change. It should NOT be used in stable production environments. -->
+      <description>Reports information for a detected Point-of-Interest (POI).
+        This can be either set by an operator (through MAV_CMD_SET_POI_FOCUS) or through onboard software or external processes.
+        The receiving system should track the POI being reported based on its UID + time of detection + time of update.
+        Detection time happens once for the same UID, while time of update changes when a specific metadata of that POI gets changed (e.g. position).
+        The time of update should be changed on the sending system, based on the determined data in regards to that specific POI.
+        So, POIs that are received again should be updated if the time of update has changed.
+        Note: The sending system should repeat the current POIs at a fixed default rate of 2Hz to keep the protocol stateless). The fixed rate though can be set by the receiver using MAV_CMD_SET_MESSAGE_INTERVAL, which is advised for POIs that are not static or for which the state updates are high - decision on the rates of some specific POIs is at the implementers consideration.
+      </description>
+      <field type="uint32_t" name="time_boot_ms" units="ms">Timestamp (time since system boot).</field>
+      <field type="uint64_t" name="time_utc_detected" units="ms" invalid="0">Timestamp (time since UNIX epoch) of the POI detection, in UTC. 0 for unknown.</field>
+      <field type="uint64_t" name="time_utc_updated" units="ms" invalid="0">Timestamp (time since UNIX epoch) of the last POI update, in UTC. 0 for unknown.</field>
+      <field type="uint8_t" name="confidence_overall" units="%" invalid="UINT8_MAX">Generic confidence level. Can be used for an implementation specific confidence level. 0..100, UINT8_MAX when unknown.</field>
+      <field type="uint8_t" name="confidence_detection" units="%" invalid="UINT8_MAX">Confidence level of the POI detection. 0..100, UINT8_MAX when unknown.</field>
+      <field type="uint8_t" name="confidence_classification" units="%" invalid="UINT8_MAX">Confidence level of the POI classification. 0..100, UINT8_MAX when unknown.</field>
+      <field type="uint8_t" name="confidence_localization" units="%" invalid="UINT8_MAX">Confidence level of the POI localization. 0..100, UINT8_MAX when unknown.</field>
```

Comment:
For all of these, where does our confidence come from? What I mean is that someone has set a POI. Is there some software that guesses what the POI is and tells us that it is "X" sure that the think is of this type at this location?

So in the case of `confidence_classification` it means some system is telling you that it has found a landing pad (say) and this is how sure it is a landing pad?
 Right?

But what are the others? E.g. `confidence_detection` - does this report how confident you are that you found any feature? So perhaps you found something and you report 80% here, but you're not sure your guess of "landing pad" , so you report 30%?

Similarly what is "POI localization". In English I would expect the word localization to mean something related to natural language translation. What does it mean?

Ditto for overall confidence, and what does it mean for "implementation specific confidence"? My guess is that this is a single measure in your confidence of the other data and that you'll let the implementer decide what is a good way to merge the other confidence values.

Reply:
> Is there some software that guesses what the POI is and tells us that it is "X" sure that the think is of this type at this location?

Correct.

> So in the case of confidence_classification it means some system is telling you that it has found a landing pad (say) and this is how sure it is a landing pad?
Right?

Right.

> But what are the others? E.g. confidence_detection - does this report how confident you are that you found any feature? So perhaps you found something and you report 80% here, but you're not sure your guess of "landing pad" , so you report 30%?

That is my assumption as well yes. This comes from the CV guys.

> Similarly what is "POI localization". In English I would expect the word localization to mean something related to natural language translation. What does it mean?

It means position.

> Ditto for overall confidence, and what does it mean for "implementation specific confidence"? My guess is that this is a single measure in your confidence of the other data and that you'll let the implementer decide what is a good way to merge the other confidence values.

That's a correct guess.